metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/AlertsProfileControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "]"));
+
+    this.mockMvc.perform(post(url).with(httpBasic(user2, password)).with(csrf())
+        .contentType(MediaType.parseMediaType("application/json;charset=UTF-8"))
+        .content(user2ProfileJson))
+        .andExpect(status().isCreated())
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user1ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "," + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user2").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
```

Comment:
I'm all for consistency, but I'm not sure I see the value in continuing a pattern that is undesirable. I'd agree that we shouldn't fix all the other tests in this PR, but these are new.